### PR TITLE
fix(a380x/pfd): visuals update & baro alt fix

### PR DESCRIPTION
**Pull request by flogross89**
## Summary of Changes

- import Arinc429Word from fbw-sdk (remove copy from `@shared`)
- match slats+flaps position dots to actual positions
- remove speed tape outline right
- remove spaces in FMA E2 cell (i.e. 1FD2 instead of 1 FD 2)
- fix LS info box position (push a little bit upwards)
- integrate PR #8235 (FMA pos. update) & #8214 (move trans alt/lvl to arinc bus)
- make tens of digital altitude readout font smaller & push to right
- altitude tape now listens to baro corrected altitude

## Screenshots (if necessary)

Before PR             | After PR (alt tape only fixed in last row)
:-------------------------:|:-------------------------:
Screenshot 2024-02-03 060109 | Screenshot 2024-02-03 055512
Screenshot 2024-02-03 060947 | Screenshot 2024-02-03 060853
Screenshot 2024-02-03 062755 | Screenshot 2024-02-03 062656

## References

https://www.youtube.com/watch?v=AI8yFhuEa4Q
various...

## Additional context

Discord username (if different from GitHub): floridude

## Testing instructions
Quite a few changes to the underlying data; Best to test a good variety of flight phases and situations

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by 2hwk — APPROVED** [review]

File: fbw-a380x/src/systems/instruments/src/PFD/LowerArea.tsx [hunk 1]
```diff
@@ -218,9 +218,9 @@ class FlapsIndicator extends DisplayComponent<{ bus: EventBus }> {
                     <path d={circlePath(0.8, 5, 196.4)} class={this.slatsTargetPos.map((i) => (i == 2 ? 'NormalStroke Stroke Fill Cyan' : 'NormalStroke White'))} />
 
                     <path d='M 32.3 193.7 v 1.7 h 1.9 z' class='Fill Stroke NormalStroke Cyan CornerRound' visibility={this.flapsTargetPos.map((i) => (i == 0 ? 'visible' : 'hidden'))} />
-                    <path d='M 39.9 196.8 v 1.7 h 1.9 z' class={this.flapsTargetPos.map((i) => (i == 1 ? 'Fill Stroke NormalStroke Cyan CornerRound' : 'Fill Stroke NormalStroke White CornerRound'))} />
-                    <path d='M 47.3 199.9 v 1.7 h 1.9 z' class={this.flapsTargetPos.map((i) => (i == 2 ? 'Fill Stroke NormalStroke Cyan CornerRound' : 'Fill Stroke NormalStroke White CornerRound'))} />
-                    <path d='M 54.7 203 v 1.7 h 1.9 z' class={this.flapsTargetPos.map((i) => (i == 3 ? 'Fill Stroke NormalStroke Cyan CornerRound' : 'Fill Stroke NormalStroke White CornerRound'))} />
+                    <path d='M 41.5 197.5 v 1.7 h 1.9 z' class={this.flapsTargetPos.map((i) => (i == 1 ? 'Fill Stroke NormalStroke Cyan CornerRound' : 'Fill Stroke NormalStroke White CornerRound'))} />
```
Comment (lukecologne) [inline fbw-a380x/src/systems/instruments/src/PFD/LowerArea.tsx:221]: We should not edit the flap/slat dots, they should be correct for the correct flaps deflections, we currently have the 320 deflections still, so they should match up once that has been fixed

**Review by lukecologne — CHANGES_REQUESTED** [review]
The flaps indicator is like this for a reason, see comment

File: fbw-a380x/src/systems/instruments/src/PFD/LowerArea.tsx (hunk 1 above)
Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/PFD/LowerArea.tsx:221]: Ok, reverted these changes.

File: fbw-a380x/src/systems/instruments/src/PFD/DigitalAltitudeReadout.tsx [hunk 2]
```diff
@@ -234,13 +234,13 @@ class Drum extends DisplayComponent<DrumProperties> {
...
-                graduationElements.push(<text ref={digitRef} transform={`translate(0 ${offset})`} class={`FontLargest MiddleAlign ${this.color}`} x="7.18" y="7.1" />);
+                graduationElements.push(<text ref={digitRef} transform={`translate(0 ${offset})`} class={`FontMedium MiddleAlign ${this.color}`} x="9.2" y="6.8" />);
             } else if (this.props.type === 'ten-thousands') {
-                graduationElements.push(<text ref={digitRef} transform={`translate(0 ${offset})`} class={`FontLargest MiddleAlign ${this.color}`} x="2.498" y="7.1" />);
+                graduationElements.push(<text ref={digitRef} transform={`translate(0 ${offset})`} class={`FontMedium MiddleAlign ${this.color}`} x="5.9" y="6.8" />);
             } else if (this.props.type === 'tens') {
-                graduationElements.push(<text ref={digitRef} transform={`translate(0 ${offset})`} class={`FontSmallest MiddleAlign ${this.color}`} x="4.5894" y="8.9133" />);
+                graduationElements.push(<text ref={digitRef} transform={`translate(0 ${offset})`} class={`FontTinyToSmallest MiddleAlign ${this.color}`} x="5.5" y="8.9133" />);
```
Comment (lukecologne) [inline fbw-a380x/src/systems/instruments/src/PFD/DigitalAltitudeReadout.tsx:243]: I did a quick check and FontSmallest is still correct for the tens digits, for the others FontMedium seems to be correct

Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/PFD/DigitalAltitudeReadout.tsx]: Ok, reverted that as well. As a side note, I also activated the gear indication, see latest commit. I think using the LGCIU input is a good trade-off, as long as we don't have the LGCIS yet.

**Comment by beheh** [comment]
Not a blocker, but the QNH indicator is also slightly different on the A380. The number is closer to the "QNH" label (and possibly both are a little smaller?).

**Comment by flogross89** [comment]
> Not a blocker, but the QNH indicator is also slightly different on the A380. The number is closer to the "QNH" label (and possibly both are a little smaller?).

Yeah, I could also touch that while I’m at it, I’m still not 100% satisfied with the kerning of the altitude readout

**Review by lukecologne — APPROVED** [review]

**Pull request merged**